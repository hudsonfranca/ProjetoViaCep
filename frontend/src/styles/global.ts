import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
*{
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Rubik', sans-serif;
    outline: none;
    text-decoration:none;
}
  body {
    min-height: 100vh;
  }
`