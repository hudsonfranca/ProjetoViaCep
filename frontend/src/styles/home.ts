import styled from "styled-components";


export const Container = styled.div`
  min-height: 100%;
  width: 100%;
  background-color:${({theme:{colors:{primary}}})=>primary};
  padding-top:100px;
  padding-bottom:100px;
`

export const Section = styled.section`
  min-height: 100vh;
  width: 100%;
  background-color:${({theme:{colors:{primary}}})=>primary};
  color:#fff;
  padding: 0 10%;
  display:flex;
  flex-direction:column;
  align-items: center;

  h1{
    font-size:2.5rem;
    margin-bottom:30px;
  }

  input{
    width: 100%;
    height: 100%;
    padding: 4px 14px;
    font-size:1.7rem;
    border: 0;
    color:#495057;
   
  }

  @media (max-width: 413px) {
    padding: 0 7%;

    input{
      font-size:1.5rem;
    }
  }
  
`

export const Form = styled.form`
  width: 100%;
  height: 81px;
  background-color:#fff;
  padding: 10px 21px;
  border-radius:0.25rem;
  display: grid;
  grid-template-columns:1fr auto;
  justify-content: center;
  align-items: center;
`

export const Button = styled.button`
  width: 111px;
  height: 61px;
  color: #fff;
  background-color:#20c997;
  border: 0 solid #20c997;
  font-size:1rem;
  font-weight:500;
  border-radius:0.25rem;
  cursor: pointer;
  transition:0.3s ease-in-out;
  display:flex;
  justify-content: center;
  align-items: center;

  &:hover{
  background-color:#02b875;
  }
  @media (max-width: 413px) {
    width: 100px;
  height: 49px
  }

`

