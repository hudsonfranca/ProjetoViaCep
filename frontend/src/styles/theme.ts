export const theme = {
  colors: {
    primary: '#4582ec',
    secondary:'#343a40',
    tertiary:'#44b976',
    quaternary:'#F0C10E'
  },
}
