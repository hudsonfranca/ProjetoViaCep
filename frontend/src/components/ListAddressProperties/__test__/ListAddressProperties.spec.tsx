
/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import {ListAddressProperties} from '../';
import { theme } from '../../../styles/theme';

const mockData =  {
  cep: "01001-000",
  logradouro: "Praça da Sé",
  complemento: "lado ímpar",
  bairro: "Sé",
  localidade: "São Paulo",
  uf: "SP",
  ibge: "3550308",
  gia: "1004",
  ddd: "11",
  siafi: "7107"
}

const defaultProps = {
  address:mockData
}

const renderComponent = (props = {}) => {
  return {
    ...render(
    <ThemeProvider theme={theme}>
      <ListAddressProperties {...defaultProps} {...props}/>
    </ThemeProvider>
  ),
    props: {
      ...defaultProps,
      ...props,
    },
  };
};

describe('<ListAddressProperties/>',()=>{

  it('shows the correct cep' ,()=>{
    renderComponent()
    expect(screen.getByText(/: 01001\-000/i)).toBeInTheDocument();
  })
  it('shows the correct logradouro' ,()=>{
    renderComponent()
    expect(screen.getByText(/logradouro: praça da sé/i)).toBeInTheDocument();
  })

  it('shows the correct complemento' ,()=>{
    renderComponent()
    expect(screen.getByText(/complemento: lado ímpar/i)).toBeInTheDocument();
  })

  it('shows the correct bairro' ,()=>{
    renderComponent()
    expect(screen.getByText(/bairro: sé/i)).toBeInTheDocument();

  })
  it('shows the correct localidade' ,()=>{
    renderComponent()
    expect(screen.getByText(/localidade: são paulo/i)).toBeInTheDocument();
  })
  it('shows the correct uf' ,()=>{
    renderComponent()
    expect(screen.getByText(/: sp/i)).toBeInTheDocument();

  })
  it('shows the correct ibge' ,()=>{
    renderComponent()
    expect(screen.getByText(/: 3550308/i)).toBeInTheDocument();

  })
  it('shows the correct gia' ,()=>{
    renderComponent()
    expect(screen.getByText(/: 1004/i)).toBeInTheDocument();

  })
  it('shows the correct ddd' ,()=>{
    renderComponent()
    expect(screen.getByText(/: 11/i)).toBeInTheDocument();

  })
  it('shows the correct siafi' ,()=>{
    renderComponent()
    expect(screen.getByText(/: 7107/i)).toBeInTheDocument();
  })
})

