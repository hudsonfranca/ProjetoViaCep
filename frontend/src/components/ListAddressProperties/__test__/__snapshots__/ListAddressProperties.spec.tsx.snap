// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListAddressProperties/> <ListAddressProperties/> Snapshot test 1`] = `
.c0 {
  height: auto;
  width: 100%;
  margin-top: 30px;
}

.c1 {
  height: 50px;
  width: 100%;
  border-bottom: 1px solid #fff;
  font-size: 1.5rem;
  padding: 26px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

@media (max-width:413px) {
  .c1 {
    font-size: 1.3rem;
  }
}

<div>
  <ul
    aria-label="Lista de propriedades do endereço"
    class="c0"
    style="padding: 20px;"
  >
    <li
      aria-label="cep"
      class="c1"
    >
      <abbr
        title="Código de Endereçamento Postal"
      >
        CEP
      </abbr>
      : 
      01001-000
    </li>
    <li
      aria-label="logradouro"
      class="c1"
    >
      Logradouro: 
      Praça da Sé
    </li>
    <li
      aria-label="complemento"
      class="c1"
    >
      Complemento: 
      lado ímpar
    </li>
    <li
      aria-label="bairro"
      class="c1"
    >
      Bairro: 
      Sé
    </li>
    <li
      aria-label="localidade"
      class="c1"
    >
      Localidade: 
      São Paulo
    </li>
    <li
      aria-label="uf"
      class="c1"
    >
      <abbr
        title="Unidade Federativa"
      >
        UF
      </abbr>
      : 
      SP
    </li>
    <li
      aria-label="ibge"
      class="c1"
    >
       
      <abbr
        title="Instituto Brasileiro de Geografia e Estatística"
      >
        IBGE
      </abbr>
      : 
      3550308
    </li>
    <li
      aria-label="gia"
      class="c1"
    >
       
      <abbr
        title="Guia de Informação e Apuração do ICMS"
      >
        GIA
      </abbr>
      : 
      1004
    </li>
    <li
      aria-label="ddd"
      class="c1"
    >
       
      <abbr
        title="Discagem direta a distância"
      >
        DDD
      </abbr>
      : 
      11
    </li>
    <li
      aria-label="siafi"
      class="c1"
    >
       
      <abbr
        title="Sistema Integrado de Administração Financeira"
      >
        SIAFI
      </abbr>
      : 
      7107
    </li>
  </ul>
</div>
`;
